### ISSUE TITLE
Keyboard shortcut to insert multiple blocks while inserter is open

### ISSUE THREAD

**Issue by jwold** [WordPress/gutenberg#16017]
TL;DR - Allow Command or Control keyboard shortcuts when combined with mouse clicks to add multiple blocks into Gutenberg while the inserter stays open. 

Recently came across an issue that I'd love to see if we could implement in Gutenberg. As a user, when I'm adding a block from the inserter, I often have a few blocks in mind that I want to try out. Right now I have to constantly open the inserter and hunt through to find blocks. Every time I add a block I have to repeat the process, often losing my place among all of the block sections. 

I'd love to hear if there are any concerns with this! The scenario I often find is that I want to add many blocks at once to Gutenberg to try and figure out exactly how I want to lay out the page. Often that means dumping blocks in, playing with them, and deleting them as needed. This shortcut would allow that and let me keep the inserter open, thus preserving the context and helping reduce cognitive load. 

Current functionality:

1. Open the inserter
2. Scroll through the inserter, expand sections, and find my block
3. Select the block I want
4. Inserter closes and I now see the block inside of my content area in Gutenberg
5. Open the inserter
6. Scroll through the inserter again to find where I was, expand the section, and find another block I want to insert
7. Select the second block
8. Inserter closes, and now I see the second block

Desired functionality:

1. Open the inserter
2. Scroll through the inserter, expand sections, and identify multiple blocks
3. On my keyboard press Command on Mac (or Control on PC) and then select the block I want with the mouse/trackpad. 
4. Inserter remains open, and my block is now inside of Gutenberg
5. Identify and select the second block while the inserter is open, press Command and select the block with mouse/trackpad. 
6. Repeat as necessary
7. Close the inserter when I’m done

With the proposed scenario the cursor would repeatedly move to the most recent block. 

From an accessibility standpoint this is unlikely to conflict with other keyboard shortcuts, as it's an isolated command tied to a mouse click. (Thanks @ryelle)

**Comment by afercia** [comment]
I'd support this feature, as long as there are no shortcuts conflicts.

This feature can be made available also to keyboard users, for example pressing `ctrl + Spacebar`.

Worth noting the same shortcut `ctrl + Spacebar` is used in the classic editor media views. It's one of the hidden features in WordPress, as in very few users and developers are aware of it. To try it:
- enable Classic Editor
- edit a post, click Add Media
- in the default "Add Media" view, users are supposed to insert one media at a time
- however, it is also possible to insert multiple media in one go
- with the mouse (on a mac): Cmd + click to select more than one media and then click "Insert into post"
- with the keyboard: navigate through the attachments with the Tab key
- press Spacebar to select one attachment
- navigate to another attachment with the Tab key
- press Ctrl + Spacebar to add the new attachment to your selection
- add as many attachments as you want
- navigate to "Insert into post" and press Enter

This works 🙂 The only downside is that it's a very hidden feature.

**Comment by jwold** [comment]
Thank you for the feedback on this! Some great points.

**Comment by youknowriad** [comment]
I think this is solved with the new design "Inserter as a panel". Thanks.
